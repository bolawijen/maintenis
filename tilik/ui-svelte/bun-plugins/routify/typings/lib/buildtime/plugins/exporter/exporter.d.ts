export function exporter({ instance }: {
    instance: RoutifyBuildtime;
}): Promise<void[]>;
export function exportNode(rootNode: RNodeBuildtime): Promise<void>;
export function exportInstance(rootNode: RNodeBuildtime): Promise<void>;
export function exportSitemap(rootNode: RNodeBuildtime): Promise<void>;
export function exportRender(instance: RoutifyBuildtime, relativePaths: RelativePaths): Promise<void>;
export function exportRoutifyInit(instance: RoutifyBuildtime, relativePaths: RelativePaths): Promise<void>;
export function exportRouteMap(instance: RoutifyBuildtime): Promise<void>;
export type RelativePaths = {
    routesDir: string;
    mainEntryPoint: string;
    rootComponent: string;
};
