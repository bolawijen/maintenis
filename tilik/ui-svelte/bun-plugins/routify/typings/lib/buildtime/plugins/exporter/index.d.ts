/** @type {RoutifyBuildtimePlugin} */
export const exporterPlugin: RoutifyBuildtimePlugin;
