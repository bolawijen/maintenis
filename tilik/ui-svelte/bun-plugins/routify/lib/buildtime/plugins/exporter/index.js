import { exporter } from './exporter.js'

/** @type {RoutifyBuildtimePlugin} */
export const exporterPlugin = {
    name: 'exporter',
    after: 'filemapper',
    build: exporter,
}
