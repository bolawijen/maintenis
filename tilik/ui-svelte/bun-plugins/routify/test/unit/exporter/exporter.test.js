import fse from 'fs-extra'
import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'
import { exportNode } from '../../../lib/buildtime/plugins/exporter/exporter.js'
import { filemapper } from '../../../lib/buildtime/plugins/filemapper/lib/index.js'
import { RoutifyBuildtime } from '../../../lib/buildtime/RoutifyBuildtime.js'

const __dirname = dirname(fileURLToPath(import.meta.url))
const options = {
    routesDir: { default: __dirname + '/example' },
}

const instance = new RoutifyBuildtime(options)

test('can build route tree', async () => {
    await filemapper({ instance })
    expect(Object.values(instance.rootNodes)[0].descendants.length).toEqual(8)
})

test('can export a route tree', async () => {
    await exportNode(Object.values(instance.rootNodes)[0], __dirname + '/temp/.routify')
    const content = fse.readFileSync(
        resolve(__dirname, 'temp/.routify', 'routes.default.js'),
        'utf-8',
    )
    expect(content).toMatchSnapshot()
})
