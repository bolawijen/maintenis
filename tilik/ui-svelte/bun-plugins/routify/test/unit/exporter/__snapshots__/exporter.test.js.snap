// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/unit/exporter/exporter.test.js/can export a route tree 1`] = `
"

export default {
  \\"meta\\": {},
  \\"id\\": \\"_default\\",
  \\"_regex\\": {},
  \\"_paramKeys\\": {},
  \\"module\\": \\"test/unit/exporter/example/_module.svelte\\",
  \\"file\\": {
    \\"path\\": \\"test/unit/exporter/example/_module.svelte\\",
    \\"dir\\": \\"test/unit/exporter/example\\",
    \\"base\\": \\"_module.svelte\\",
    \\"ext\\": \\".svelte\\",
    \\"name\\": \\"_module\\"
  },
  \\"rootName\\": \\"default\\",
  \\"children\\": [
    {
      \\"meta\\": {
        \\"reset\\": true
      },
      \\"id\\": \\"_default_admin\\",
      \\"_regex\\": {},
      \\"_paramKeys\\": {},
      \\"name\\": \\"admin\\",
      \\"module\\": \\"test/unit/exporter/example/admin/_reset.svelte\\",
      \\"file\\": {
        \\"path\\": \\"test/unit/exporter/example/admin/_reset.svelte\\",
        \\"dir\\": \\"test/unit/exporter/example/admin\\",
        \\"base\\": \\"_reset.svelte\\",
        \\"ext\\": \\".svelte\\",
        \\"name\\": \\"_reset\\"
      },
      \\"children\\": [
        {
          \\"meta\\": {},
          \\"id\\": \\"_default_admin_index_svelte\\",
          \\"_regex\\": {},
          \\"_paramKeys\\": {},
          \\"name\\": \\"index\\",
          \\"module\\": \\"test/unit/exporter/example/admin/index.svelte\\",
          \\"file\\": {
            \\"path\\": \\"test/unit/exporter/example/admin/index.svelte\\",
            \\"dir\\": \\"test/unit/exporter/example/admin\\",
            \\"base\\": \\"index.svelte\\",
            \\"ext\\": \\".svelte\\",
            \\"name\\": \\"index\\"
          },
          \\"children\\": []
        },
        {
          \\"meta\\": {},
          \\"id\\": \\"_default_admin_page_svelte\\",
          \\"_regex\\": {},
          \\"_paramKeys\\": {},
          \\"name\\": \\"page\\",
          \\"module\\": \\"test/unit/exporter/example/admin/page.svelte\\",
          \\"file\\": {
            \\"path\\": \\"test/unit/exporter/example/admin/page.svelte\\",
            \\"dir\\": \\"test/unit/exporter/example/admin\\",
            \\"base\\": \\"page.svelte\\",
            \\"ext\\": \\".svelte\\",
            \\"name\\": \\"page\\"
          },
          \\"children\\": []
        }
      ]
    },
    {
      \\"meta\\": {},
      \\"id\\": \\"_default_blog\\",
      \\"_regex\\": {},
      \\"_paramKeys\\": {},
      \\"name\\": \\"blog\\",
      \\"module\\": false,
      \\"file\\": {
        \\"path\\": \\"test/unit/exporter/example/blog\\",
        \\"dir\\": \\"test/unit/exporter/example\\",
        \\"base\\": \\"blog\\",
        \\"ext\\": \\"\\",
        \\"name\\": \\"blog\\"
      },
      \\"children\\": [
        {
          \\"meta\\": {
            \\"dynamic\\": true
          },
          \\"id\\": \\"_default_blog__slug__svelte\\",
          \\"_regex\\": {},
          \\"_paramKeys\\": {},
          \\"name\\": \\"[slug]\\",
          \\"module\\": \\"test/unit/exporter/example/blog/[slug].svelte\\",
          \\"file\\": {
            \\"path\\": \\"test/unit/exporter/example/blog/[slug].svelte\\",
            \\"dir\\": \\"test/unit/exporter/example/blog\\",
            \\"base\\": \\"[slug].svelte\\",
            \\"ext\\": \\".svelte\\",
            \\"name\\": \\"[slug]\\"
          },
          \\"children\\": []
        },
        {
          \\"meta\\": {
            \\"dynamic\\": true
          },
          \\"id\\": \\"_default_blog_prefixed_slug__svelte\\",
          \\"_regex\\": {},
          \\"_paramKeys\\": {},
          \\"name\\": \\"prefixed[slug]\\",
          \\"module\\": \\"test/unit/exporter/example/blog/prefixed[slug].svelte\\",
          \\"file\\": {
            \\"path\\": \\"test/unit/exporter/example/blog/prefixed[slug].svelte\\",
            \\"dir\\": \\"test/unit/exporter/example/blog\\",
            \\"base\\": \\"prefixed[slug].svelte\\",
            \\"ext\\": \\".svelte\\",
            \\"name\\": \\"prefixed[slug]\\"
          },
          \\"children\\": []
        }
      ]
    },
    {
      \\"meta\\": {},
      \\"id\\": \\"_default_index_svelte\\",
      \\"_regex\\": {},
      \\"_paramKeys\\": {},
      \\"name\\": \\"index\\",
      \\"module\\": \\"test/unit/exporter/example/index.svelte\\",
      \\"file\\": {
        \\"path\\": \\"test/unit/exporter/example/index.svelte\\",
        \\"dir\\": \\"test/unit/exporter/example\\",
        \\"base\\": \\"index.svelte\\",
        \\"ext\\": \\".svelte\\",
        \\"name\\": \\"index\\"
      },
      \\"children\\": []
    },
    {
      \\"meta\\": {
        \\"dynamic\\": true,
        \\"dynamicSpread\\": true
      },
      \\"_regex\\": {},
      \\"_paramKeys\\": {},
      \\"name\\": \\"[...404]\\",
      \\"module\\": \\"./.routify/components/[...404].svelte\\",
      \\"file\\": {
        \\"path\\": \\".routify/components/[...404].svelte\\",
        \\"dir\\": \\".routify/components\\",
        \\"base\\": \\"[...404].svelte\\",
        \\"ext\\": \\".svelte\\",
        \\"name\\": \\"[...404]\\"
      },
      \\"children\\": []
    }
  ]
}"
`;
